### PR TITLE
Fix deprecation warnings for sklearn/scipy

### LINKED ISSUE #606: Remove the `except` part and use `n_features_in_` everywhere
Related to https://github.com/microsoft/hummingbird/pull/605#discussion_r916084923

### PR DESCRIPTION
**Pull request by mshr-h**
Fix #606 and more

- Since version 1.0,  it is not needed to import enable_hist_gradient_boosting anymore
  - [sklearn.experimental.enable_hist_gradient_boosting — scikit-learn 1.1.2 documentation](https://scikit-learn.org/stable/modules/generated/sklearn.experimental.enable_hist_gradient_boosting.html)
- `sigma_` is deprecated in 1.0 and will be removed in 1.2. Use `var_` instead.
  - [sklearn.naive_bayes.GaussianNB — scikit-learn 1.1.2 documentation](https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.GaussianNB.html)
- FutureWarning: algorithm='auto' is deprecated, it will be removed in 1.3. Using 'lloyd' instead.
  - [sklearn.cluster.k_means — scikit-learn 1.1.2 documentation](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.k_means.html)
- FutureWarning: algorithm='full' is deprecated, it will be removed in 1.3. Using 'lloyd' instead.
  - [sklearn.cluster.k_means — scikit-learn 1.1.2 documentation](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.k_means.html)
- DeprecationWarning: Please use `csr_matrix` from the `scipy.sparse` namespace, the `scipy.sparse.csr` namespace is deprecated.
  - [scipy.sparse.csr_matrix — SciPy v1.9.0 Manual](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_matrix.html)
- FutureWarning: The loss 'log' was deprecated in v1.1 and will be removed in version 1.3. Use `loss='log_loss'` which is equivalent.
  - [sklearn.linear_model.SGDClassifier — scikit-learn 1.1.2 documentation](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html)
- FutureWarning: Attribute `lambdas_` was deprecated in version 1.0 and will be removed in 1.2. Use `eigenvalues_` instead.
  - [sklearn.decomposition.KernelPCA — scikit-learn 1.1.2 documentation](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.KernelPCA.html)
- FutureWarning: Attribute `alphas_` was deprecated in version 1.0 and will be removed in 1.2. Use `eigenvectors_` instead.
  - [sklearn.decomposition.KernelPCA — scikit-learn 1.1.2 documentation](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.KernelPCA.html)

### CODE REVIEW

**Comment by mshr-h** [comment]
Should be good.

@ksaur @interesaaat 
Can you review it?

**Comment by mshr-h** [comment]
Should be good.

@ksaur @interesaaat 
Can you review it?

**Review by ksaur — APPROVED** [review]
Thank you!!
